src/gleam/string.gleam

```diff
@@ -396,13 +397,15 @@ pub fn trim_right(string: String) -> String {
 //
 // ///
 // pub fn to_graphemes(string: String) -> List(String) {}
-// TODO
-// Split a non-empty string into its head and tail. This lets you
-// pattern match on strings exactly as you would with lists.
-//
-// ## Examples
-//    > next_grapheme("")
-//    Error(Nil)
-//
-//
-// pub fn next_grapheme(string: String) -> Option(tuple(Grapheme, String)) {}
+/// Split a non-empty string into its head and tail. This lets you
+/// pattern match on strings exactly as you would with lists.
+///
+/// ## Examples
+///    > next_grapheme("gleam")
+///    Ok(tuple("g", "leam"))
+///
+///    > next_grapheme("")
+///    Error(Nil)
+///
+pub external fn next_grapheme(string: String) -> Option(tuple(String, String)) =
+  "gleam_stdlib" "string_next_grapheme"
```

Comment:
Great! Thank you!

One question: Does `pop_grapheme` make more sense as the name here? Perhaps `next_grapheme` returns the grapheme without the rest of the binary?

Reply:
Hmmm, I don't know, 'pop' sounds a bit mutable to me.
Right now it's exactly like: https://hexdocs.pm/elixir/String.html#next_grapheme/1

It actually seems to be the same as Elm's [uncons](https://package.elm-lang.org/packages/elm/core/latest/String#uncons) (as we don't have a seperate Char type)

`uncons` also pops up when I do a [Hoogle search for this type signature](https://hoogle.haskell.org/?hoogle=String%20-%3E%20Maybe%20(String%2C%20String))

Reply:
Let's stick with next then

Reply:
I had a look around and Elixir uses `pop` for access, map, dict, and keyword, String seems to be the odd one out.

Thinking about this operation on other data structures such as lists, queues, etc, I think pop makes more sense.

I would like to avoid uncons as it's quite niche.

Reply:
`uncons` sounds a bit nich indeed. I'll change it to `pop_grapheme`.